c10/cuda/CUDAMiscFunctions.cpp

```diff
@@ -1,12 +1,14 @@
 #include <c10/cuda/CUDAMiscFunctions.h>
-#include <cstdlib>
+#include <c10/util/env.h>
 
 namespace c10::cuda {
 
+// NOLINTNEXTLINE(bugprone-exception-escape,-warnings-as-errors)
 const char* get_cuda_check_suffix() noexcept {
-  static char* device_blocking_flag = getenv("CUDA_LAUNCH_BLOCKING");
+  static auto device_blocking_flag =
+      c10::utils::check_env("CUDA_LAUNCH_BLOCKING");
   static bool blocking_enabled =
-      (device_blocking_flag && atoi(device_blocking_flag));
+      (device_blocking_flag.has_value() && device_blocking_flag.value());
```

Comment:
This doesn't look semantics preserving

Reply:
+1 but do we think this is the likely culprit (seems rare to see anything outside of `CUDA_LAUNCH_BLOCKING=0` or `CUDA_LAUNCH_BLOCKING=1`

Reply:
check_env assumes the value is "0" or "1"  and then coverts it to bool. So it's semantics preserving.